### PR TITLE
feat(ui): implement auth flow

### PR DESCRIPTION
**Pull request by steveninhle**
- Created auth context for the app
- Implement login with Email
- Implement login with Google
- Implement form validation with @hookform

image

image

image

### CODE REVIEW

File: ui/dashboard/src/pages/signin/signin.tsx [hunk 1]
```diff
@@ -33,15 +36,20 @@ const SignIn = () => {
         <h1 className="text-gray-900 typo-head-bold-huge">
           {`Sign in to Bucketeer`}
         </h1>
-        {/* <p className="text-accent-red-500 typo-para-medium">
-          No account found for this Google Account. Please check the email
-          entered or try a different Google Account.
-        </p> */}
+        {isGoogleAuthError && (
+          <p className="text-accent-red-500 typo-para-medium">
+            {` No account found for this Google Account. Please check the email
```
Comment (cre8ivejp) [inline ui/dashboard/src/pages/signin/signin.tsx:41]: Let's implement the translation assets.
For now, please use English for both English and Japanese translations.

File: ui/dashboard/src/pages/signin/signin.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        <Button
+          variant="secondary-2"
+          onClick={() => {
+            navigate(PAGE_PATH_AUTH_SIGNIN);
+            setIsGoogleAuthError(false);
+          }}
+        >
+          <Icon icon={IconEmail} />
+          {`Sign in With Email`}
+        </Button>
```
Comment (cre8ivejp) [inline ui/dashboard/src/pages/signin/signin.tsx:56]: This button must only show when the `DEMO_SIGN_IN_ENABLED`.

Reference.
https://github.com/bucketeer-io/bucketeer/blob/main/ui/web-v2/src/pages/auth/signin.tsx#L30-L37

Also, please check `DEMO_SIGN_IN_ENABLED` in the root http://localhost:8000/auth/signin if the user tries to access it directly. We should redirect to the login page if the user is not logged in or the 404 page if it is logged in.

File: ui/dashboard/src/pages/signin/email.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        <p>{`Email: demo@bucketeer.io`}</p>
+        <p>{`Password: demo`}</p>
+      </div>
+
+      {showAuthError && (
+        <p className="text-accent-red-500 typo-para-medium mt-6">
+          {`Wrong email or password. Try again or`}
+          <Link to={PAGE_PATH_ROOT}>
+            <span className="underline ml-1">{`create an account.`}</span>
+          </Link>
```
Comment (cre8ivejp) [inline ui/dashboard/src/pages/signin/email.tsx:98]: Wrong email or password. Try again or
```suggestion
          {`Invalid email or password.`}
```

File: ui/dashboard/src/components/ellipsis-spinner/index.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import styles from './styles.module.css';
+
+const EllipsisSpinner = () => {
+  return (
+    <div className={styles.main}>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
```
Comment (cre8ivejp) [inline ui/dashboard/src/components/ellipsis-spinner/index.tsx:9]: What is this for?

Comment (steveninhle) [reply ui/dashboard/src/components/ellipsis-spinner/index.tsx:9]: These elements create the loading. Do you like the loading like this? or can we only use spinner loading?

https://github.com/user-attachments/assets/2fc8a2b0-8a03-43c5-a283-93d101ff9ec8

Comment (steveninhle) [reply ui/dashboard/src/components/ellipsis-spinner/index.tsx:9]: https://github.com/user-attachments/assets/bc9c532e-a93b-4ecd-abe3-68c7a930e040

Comment (cre8ivejp) [reply ui/dashboard/src/components/ellipsis-spinner/index.tsx:9]: @steveninhle, the second one, please.

File: ui/dashboard/public/locales/ja/auth.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  "email": "Email",
+  "password": "Password",
+  "first-name": "First Name",
+  "first-name-placeholder": "Enter your first name",
+  "last-name": "Last Name",
+  "last-name-placeholder": "Enter your last name",
+  "organization": "Organization",
+  "organization-placeholder": "Select your Organization",
+  "sign-in": {
+    "description": "To access our Demo site, please sign in using the follow in information."
```
Comment (cre8ivejp) [inline ui/dashboard/public/locales/ja/auth.json:11]:
```suggestion
    "description": "To access our Demo site, please sign in using the following information."
```

File: ui/dashboard/public/locales/en/auth.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "last-name": "Last Name",
+  "last-name-placeholder": "Enter your last name",
+  "organization": "Organization",
+  "organization-placeholder": "Select your Organization",
+  "sign-in": {
+    "description": "To access our Demo site, please sign in using the follow in information."
+  },
+  "select-organization": {
+    "title": "Select your Organization",
+    "description": "Select the organization you want to work for."
```
Comment (cre8ivejp) [inline ui/dashboard/public/locales/en/auth.json:15]:
```suggestion
    "description": "Select the organization you want to access."
```

File: ui/dashboard/public/locales/en/auth.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  "email": "Email",
+  "password": "Password",
+  "first-name": "First Name",
+  "first-name-placeholder": "Enter your first name",
+  "last-name": "Last Name",
+  "last-name-placeholder": "Enter your last name",
+  "organization": "Organization",
+  "organization-placeholder": "Select your Organization",
+  "sign-in": {
+    "description": "To access our Demo site, please sign in using the follow in information."
```
Comment (cre8ivejp) [inline ui/dashboard/public/locales/en/auth.json:11]:
```suggestion
    "description": "To access our Demo site, please sign in using the following information."
```

File: ui/dashboard/public/locales/ja/auth.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "select-organization": {
+    "title": "Select your Organization",
+    "description": "Select the organization you want to work for."
```
Comment (cre8ivejp) [inline ui/dashboard/public/locales/ja/auth.json:15]:
```suggestion
    "description": "Select the organization you want to access."
```

File: ui/dashboard/.env.example [hunk 9]
```diff
@@ -1,2 +1,3 @@
-DEV_WEB_API_ENDPOINT=https://example.com
-DEV_AUTH_REDIRECT_ENDPOINT=http://localhost:8000
+VITE_RELEASE_CHANNEL=dev
+VITE_WEB_API_ENDPOINT=https://example.com
+VITE_AUTH_REDIRECT_ENDPOINT=http://localhost:8000
```
Comment (cre8ivejp) [inline ui/dashboard/.env.example:3]:
```suggestion
VITE_RELEASE_CHANNEL=dev
VITE_WEB_API_ENDPOINT=https://example.com
VITE_AUTH_REDIRECT_ENDPOINT=http://localhost:8000
VITE_DEMO_SIGN_IN_ENABLED=false
VITE_DEMO_SIGN_IN_EMAIL=
VITE_DEMO_SIGN_IN_PASSWORD=
```

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**